runelite-client/src/main/java/net/runelite/client/plugins/zoneindicators/ZoneIndicatorsPlugin.java

```diff
@@ -0,0 +1,296 @@
+
+		if (isOpenableAt(wa1.toWorldPoint()) || isOpenableAt(wa2.toWorldPoint()))
+		{
+			// When there's something with the open option (e.g. a door) on the tile,
+			// we assume it can be opened and walked through afterwards. Without this
+			// check, the line for that tile wouldn't render with collision detection
+			// because the collision check isn't done if collision data changes.
+			return true;
+		}
+
+		boolean b1 = wa1.canTravelInDirection(client, -dy, -dx);
+		boolean b2 = wa2.canTravelInDirection(client, dy, dx);
+		return b1 && b2;
+	}
+
+	private void findLinesInScene()
+	{
+		inDeadman = client.getWorldType().stream().anyMatch(x ->
+			x == WorldType.DEADMAN || x == WorldType.SEASONAL_DEADMAN);
+		inPvp = client.getWorldType().stream().anyMatch(x ->
+			x == WorldType.PVP || x == WorldType.PVP_HIGH_RISK);
+
+		Rectangle sceneRect = new Rectangle(
+			client.getBaseX() + 1, client.getBaseY() + 1,
+			Constants.SCENE_SIZE - 2, Constants.SCENE_SIZE - 2);
+
+		// Generate lines for multicombat zones
+		if (config.multicombatZoneVisibility() == ZoneVisibility.HIDE)
+		{
+			for (int i = 0; i < multicombatPathToDisplay.length; i++)
+			{
+				multicombatPathToDisplay[i] = null;
+			}
+		}
+		else
+		{
+			for (int i = 0; i < multicombatPathToDisplay.length; i++)
+			{
+				currentPlane = i;
+
+				GeneralPath lines = new GeneralPath(MapLocations.getMulticombat(sceneRect, i));
+				lines = Geometry.clipPath(lines, sceneRect);
+				if (config.multicombatZoneVisibility() == ZoneVisibility.SHOW_IN_PVP &&
+					!isInDeadman() && !isInPvp())
+				{
+					lines = Geometry.clipPath(lines, MapLocations.getRoughWilderness(i));
+				}
+				lines = Geometry.unitifyPath(lines, 1);
+				if (config.collisionDetection())
+				{
+					lines = Geometry.filterPath(lines, this::collisionFilter);
+				}
+				lines = Geometry.transformPath(lines, this::transformWorldToLocal);
+				multicombatPathToDisplay[i] = lines;
+			}
+		}
+
+		// Generate safezone lines for deadman/pvp worlds
+		for (int i = 0; i < pvpPathToDisplay.length; i++)
+		{
+			currentPlane = i;
+
+			GeneralPath safeZonePath = null;
+			if (config.showDeadmanSafeZones() && isInDeadman())
+			{
+				safeZonePath = new GeneralPath(MapLocations.getDeadmanSafeZones(sceneRect, i));
+			}
+			else if (config.showPvpSafeZones() && isInPvp())
+			{
+				safeZonePath = new GeneralPath(MapLocations.getPvpSafeZones(sceneRect, i));
+			}
+			if (safeZonePath != null)
+			{
+				safeZonePath = Geometry.clipPath(safeZonePath, sceneRect);
+				safeZonePath = Geometry.unitifyPath(safeZonePath, 1);
+				if (config.collisionDetection())
+				{
+					safeZonePath = Geometry.filterPath(safeZonePath, this::collisionFilter);
+				}
+				safeZonePath = Geometry.transformPath(safeZonePath, this::transformWorldToLocal);
+			}
+			pvpPathToDisplay[i] = safeZonePath;
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getKey().equals("collisionDetection") ||
```

Comment:
"collisionDetection".equals(event.getKey()) would be better null handling

Reply:
ConfigChanged is always posted with a key so it doesn't matter in this case.